// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.helloworld";
option java_outer_classname = "HelloWorldProto";
option objc_class_prefix = "HLW";

package helloworld;

// The greeting service definition.
service Greeter {
  // Sends a greeting
  rpc Cadastra (CadastrarCli_Request) returns (CadastrarCli_Reply) {}

  rpc ObtemPorCpf (ObterCliCPF_Request) returns (ObterCli_Reply)  {}

  rpc ObtemPorID (ObterCliID_Request) returns (ObterCli_Reply) {}
}

// The request message containing the user's name.
message CadastrarCli_Request {
  string nome = 1;
  int64 cpf = 2;
  string endereco = 3;
}

// The response message containing the greetings
message CadastrarCli_Reply {
  int32 id = 1;
}

message ObterCliCPF_Request {
  int64 cpf = 1;
}

message ObterCliID_Request {
  int32 id = 1;
}

message ObterCli_Reply {
  string nome = 1;
  int64 cpf = 2;
  string endereco = 3;
  int32 id = 4;
}